spacy/lang/en/__init__.py

```diff
@@ -46,4 +25,24 @@ class English(Language):
     Defaults = EnglishDefaults
 
 
+@English.factory(
+    "lemmatizer",
+    assigns=["token.lemma"],
+    default_config={"model": None, "mode": "rule", "lookups": None},
+    scores=["lemma_acc"],
+    default_score_weights={"lemma_acc": 1.0},
+)
+def make_lemmatizer(
+    nlp: Language,
+    model: Optional[Model],
+    name: str,
+    mode: str,
+    lookups: Optional[Lookups],
+):
+    tables = ["lemma_lookup", "lemma_rules", "lemma_exc", "lemma_index"]
+    if lookups is None:
+        lookups = load_lookups(lang=nlp.lang, tables=tables)
```

Comment:
Now that we have the lemmatizer set up as a component, we could consider making `load_lookups` strict in the lemmatizers if `mode` is set to `"rule"` or `"lookup"` and no custom lookups are provided?

If the user doesn't have `spacy-lookups-data` installed, they'd see a useful error or at least a warning telling them to install it or provide their own rules because otherwise, there won't be any lemmas.

Reply:
I think it's currently strict by default? I definitely saw errors about missing tables here if `spacy-lookups-data` wasn't installed.

Reply:
Ahh sorry, I think I remembered the default wrong. In that case, we could probably do something like this?

```python
if lookups is None:
    strict = mode in ("rule", "lookup")
    lookups = load_lookups(lang=nlp.lang, tables=tables, strict=strict)
```

And then have `load_lookups` actually raise if `spacy_lookups_data` isn't installed.

Reply:
I tried to redo it like this and be more specific about the tables for each mode, but it still feels a little weird overall. I think it might be better to just not load any tables for modes that `make_lemmatizer` doesn't know about? I also changed the mode setting to make it immutable, which means you can't switch modes on the fly, but it makes the initialization a lot cleaner.

Reply:
I think it's fine to ignore the tables but maybe we could raise if you add a lemmatizer to the pipeline with mode `rule` or `lookup` for a language that we don't have any data for? Then the user gets instant feedback that lemmatization isn't going to produce any results out-of-the-box and that they need to provide their own function that returns lookup tables.

> I also changed the mode setting to make it immutable, which means you can't switch modes on the fly, but it makes the initialization a lot cleaner.

Yeah, that makes sense!